cachix/src/Cachix/Client/Commands.hs

```diff
@@ -280,6 +281,11 @@ pushStorePath env config name storePath = do
               Right NoContent -> return ()
 
 
+-- TODO: should one error abort the whole pushing process? Or do a retry? Or keep going and then fail?
+eitherToPanic :: Show err => Either err val -> val
```

Comment:
You don't need an exception in pure code to do this. Three lines up on the use site you can throw a proper IO exception